### PR TITLE
UCP: fix error handing on wireup

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
Cherry-picked from https://github.com/openucx/ucx/pull/1475 + refactoring

@yosefe pls take a look, i addressed your comments regarding to the fix here

### CODE REVIEW

**Comment by yosefe** [comment]
ok to test

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3972/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3974/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

File: src/ucp/wireup/stub_ep.c [hunk 1]
```diff
@@ -416,3 +429,19 @@ int ucp_stub_ep_test(uct_ep_h uct_ep)
...
+}
+
+void ucp_stub_ep_extract_aux(uct_ep_h *uct_ep)
+{
+    uct_ep_h ep            = *uct_ep;
+    ucp_stub_ep_t *stub_ep = ucs_derived_of(ep, ucp_stub_ep_t);
+
+    *uct_ep = stub_ep->aux_ep;
+    stub_ep->aux_ep = NULL;
+    uct_ep_destroy(ep);
```
Comment (yosefe) [inline src/ucp/wireup/stub_ep.c]: this should be done by the caller - stub_ep should just extract aux_ep, not destroy user's ep

File: src/ucp/core/ucp_worker.c [hunk 2]
```diff
@@ -303,6 +292,13 @@ ucp_worker_iface_error_handler(void *arg, uct_ep_h uct_ep, ucs_status_t status)
         ucp_ep->uct_eps[failed_lane] = NULL;
     }
 
+    /* NOTE: if failed ep is stub auxiliary then we need to replace the lane
+     *       with failed ep and destroy stub ep
+     */
+    if (ucp_stub_ep_test(ucp_ep_iter->uct_eps[0])) {
```
Comment (yosefe) [inline src/ucp/core/ucp_worker.c]: shouldn't it be ucp_stub_ep_test_aux() ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3991/ for details (Mellanox internal link).

File: src/ucp/core/ucp_worker.c (hunk 2 above)
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_worker.c]: It will be over check but i'll fix for better readability

Comment (yosefe) [inline src/ucp/core/ucp_worker.c]: thanks

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4003/ for details (Mellanox internal link).

**Pull request merged**